src/vos/tests/evt_ctl.sh

```diff
@@ -166,7 +157,12 @@ cmd+=" -b -2 -D"
 echo "$cmd"
 
 $cmd -t
+result="${PIPESTATUS[0]}"
 
-result=$?
-echo Test returned $result
-exit $result
+if [[ -z ${CMOCKA_MESSAGE_OUTPUT} || -z ${CMOCKA_XML_FILE} ]]; then
+	echo "Test returned $result"
+	exit "$result"
+else
+	echo "Test complete"
+	exit 0
+fi
```

Comment:
Ahh.  Wait.  I see what you are trying to achieve here, but this is one step in the chain too high.

It's `evt_ctl` that should be altering it's exit status (0 if it ran successfully regardless of tests fail/pass if XML output is enabled and 1 if tests failed if XML output is disabled) based on whether XML output is being generated or not.  It should not be left up to a wrapper script to effect that behaviour because somebody could run it without a wrapper, or with a different wrapper and they should get the same behaviour.

Why doesn't `btree.sh` have or need this same code in it's wrapper to effect the same behaviour?

Reply:
The evt_ctl.c and btree C files are already modified to CMOCKA format. CMOCKA doesn't return values. I mentioned this information during one of our skype conversation. CMOCKA framework w has void return values since all output are stored in the XML files. Presently, $result will be zero. I didn't want to remove it till we have good runs with the above changes. If needed, I can create a minor ticket and remove it once we have some test cycles.

Reply:
If needed, we can remove the whole CMOCKA env test code (since CMOCKA C code will not return anything)